packages/devtools_app/lib/src/screens/profiler/panes/bottom_up.dart

```diff
@@ -55,10 +66,38 @@ class CpuBottomUpTable extends StatelessWidget {
       'bottom up tree).';
 
   static const selfTimeTooltip =
-      'For top-level methods in the bottom-up tree (leaf stack frames in the '
-      'CPU profile),\nthis is the time the method spent executing only its own '
-      'code. For sub nodes (the\ncallers in the CPU profile), this is the self '
-      'time of the callee when being called by\nthe caller. ';
+      'For top-level methods in the bottom-up tree (stack frames that were at '
+      'the top of at least one CPU sample), this is the time the method spent '
+      'executing only its own code.\mFor children methods in the bottom-up '
+      'tree (the callers), this is the self time of the top-level method (the '
+      'callee) when called through the child method (the caller).';
+
+  static InlineSpan? _bottomUpTimeTooltipBuilder(
+    String type,
+    CpuStackFrame stackFrame,
+    BuildContext context,
+  ) {
+    // TODO(kenz): consider adding a tooltip for root nodes as well if this is
+    // a point of confusion for the user.
+    if (stackFrame.isRoot) {
+      return null;
+    }
+    final fixedStyle = Theme.of(context).tooltipFixedFontStyle;
+    return TextSpan(
+      children: [
+        TextSpan(text: '$type time for '),
+        TextSpan(
+          text: '${stackFrame.root.name}\n',
+          style: fixedStyle,
+        ),
+        const TextSpan(text: 'when called through '),
+        TextSpan(
+          text: '${stackFrame.name}',
```

Comment:
Are we using this syntax because `stackFrame.name` can be `null`? If not, just use `stackFrame.name` directly.